deps/cares/src/ares_gethostbyname.c

```diff
@@ -188,8 +188,9 @@ static void host_callback(void *arg, int status, int timeouts,
       else if (hquery->sent_family == AF_INET6)
         {
           status = ares_parse_aaaa_reply(abuf, alen, &host, NULL, NULL);
-          if ((status == ARES_ENODATA || status == ARES_EBADRESP) &&
-               hquery->want_family == AF_UNSPEC) {
+          if ((status == ARES_ENODATA || status == ARES_EBADRESP ||
+               (status == ARES_SUCCESS && host && host->h_addr_list[0] == NULL)) &&
```

Comment:
Just recalling, are we going to switch io.js to use the proper error names instead of the made up one we have (`ENOTFOUND`)?